### PR TITLE
WIP: Update to5.17.2

### PR DESCRIPTION
**Pull request by fmeeuw**
Wait with merging until dependencies have also been updated to 5.17.2. It can be reviewed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ sudo: false
 branches:
   only:
     - master
+    - updateTo5.17.2
```
Comment (ktatarnikov) [inline .travis.yml]: dont forget to drop updateTo5.17.2

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,19 @@
 Changes
 =======
 
-# 5.15.0 / UNRELEASED
+# 5.16.0 / 07-26-2017
```
Comment (ktatarnikov) [inline CHANGELOG.md:4]: Is this actually 5.16 or 5.17? Not sure why datadog has no changes for 5.17

File: packaging/datadog-agent/source/setup_agent.sh [hunk 3]
```diff
@@ -11,7 +11,7 @@ set -u
 # SCRIPT KNOBS
 #######################################################################
 # Update for new releases, will pull this tag in the repo
-DEFAULT_AGENT_VERSION="5.14.1"
+DEFAULT_AGENT_VERSION="5.16.0"
```
Comment (ktatarnikov) [inline packaging/datadog-agent/source/setup_agent.sh]: if it upgrade to 5.17 why it says everywhere 5.16?

File: stsagent.py [hunk 4]
```diff
@@ -272,7 +272,7 @@ def on_response(self, response):
         if response.error:
             log.error("Response: %s" % response)
             if response.code in RESPONSES_TO_REJECT:
-                self._trManager.tr_error_reject_request(self)
+                self._trManager.tr_error_reject_request(self, response.code)
```
Comment (ktatarnikov) [inline stsagent.py:275]: "_trManager" formatted not according to PEP8, and of course don't change it if it is coming from upstream like that

File: utils/kubernetes/kubeutil.py [hunk 5]
```diff
@@ -459,23 +549,32 @@ def get_node_hosttags(self):
...
             if not node_name:
                 raise ValueError("node name missing or empty")
             request_url = "%s/nodes/%s" % (self.kubernetes_api_url, node_name)
-            node_info = self.retrieve_json_auth(request_url)
+            node_info = self.retrieve_json_auth(request_url).json()
             version = node_info.get("status").get("nodeInfo").get("kubeletVersion")
             tags.append("kubelet_version:%s" % version[1:])
+
+            node_labels = node_info.get('metadata', {}).get('labels', {})
+            for l_name, t_name in self.kube_node_labels.iteritems():
```
Comment (ktatarnikov) [inline utils/kubernetes/kubeutil.py:574]: is this our code? could you please rename l_name to label_name or just label.

File: utils/kubernetes/kubeutil.py [hunk 6]
```diff
@@ -174,7 +150,8 @@ def __init__(self, **kwargs):
         self.init_retry_interval = init_config.get('init_retry_interval', DEFAULT_RETRY_INTERVAL)
         self.last_init_retry = None
         self.left_init_retries = init_config.get('init_retries', DEFAULT_INIT_RETRIES) + 1
-        self.init_kubelet(instance)
+        if 'use_kubelet' not in kwargs or kwargs['use_kubelete']:
```
Comment (ktatarnikov) [inline utils/kubernetes/kubeutil.py]: Looks like it use_kubelet != use_kubelete

File: utils/kubernetes/kubeutil.py [hunk 7]
```diff
@@ -150,7 +150,7 @@ def __init__(self, **kwargs):
         self.init_retry_interval = init_config.get('init_retry_interval', DEFAULT_RETRY_INTERVAL)
         self.last_init_retry = None
         self.left_init_retries = init_config.get('init_retries', DEFAULT_INIT_RETRIES) + 1
-        if 'use_kubelet' not in kwargs or kwargs['use_kubelete']:
+        if 'use_kubelet' not in kwargs or kwargs['use_kubelet']:
```
Comment (ktatarnikov) [inline utils/kubernetes/kubeutil.py]: looks like it fixed, please ignore previous comment...

**Review by ktatarnikov — CHANGES_REQUESTED** [review]
please take a look to my minor comments...

File: utils/kubernetes/kubeutil.py (hunk 5 above)
Comment (fmeeuw) [reply utils/kubernetes/kubeutil.py:574]: No this is not our code, so I don't think it is worth it to deviate from their code for this.

File: stsagent.py (hunk 4 above)
Comment (fmeeuw) [reply stsagent.py:275]: unfortunately not our code as well.

File: packaging/datadog-agent/source/setup_agent.sh (hunk 3 above)
Comment (fmeeuw) [reply packaging/datadog-agent/source/setup_agent.sh]: Good point, I don't know why they didn't update this version....
I think we should change it to 5.17.2, which seems a better default for us.

File: CHANGELOG.md (hunk 2 above)
Comment (fmeeuw) [reply CHANGELOG.md:4]: They updated the changelog afterwards... doh. I don't think we should bother with updating this though.

**Pull request merged**